File: macOS SDK v4 Sample/DeactivateViewController.m
//
//  DeactivateViewController.m
//  macOS SDK v4 Sample
//
//  Created by Louis Harwood on 30/11/2020.
//  Copyright © 2020 Paddle.com LTD. All rights reserved.
//

#import "DeactivateViewController.h"
#import "AppConfig.h"
@import Paddle;

@interface DeactivateViewController () <PADProductDelegate>

@end

@implementation DeactivateViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do view setup here.
    if (![[self mainProduct] activated]) {
        [[self.productSelection itemAtIndex:0] setEnabled:NO];
    }
    
    if (![[self secondProduct] activated]) {
        [[self.productSelection itemAtIndex:1] setEnabled:NO];
    }
}

- (IBAction)cancelButton:(id)sender {
    [self.view.window close];
}

- (IBAction)nextButton:(id)sender {
    if (self.productSelection.indexOfSelectedItem == 0) {
        [Paddle.sharedInstance showLicenseActivationDialogForProduct:[self mainProduct] email:nil licenseCode:nil activationStatusCompletion:^(PADActivationState activationState) {
            [self.view.window close];
        }];
    } else {
        [Paddle.sharedInstance showLicenseActivationDialogForProduct:[self secondProduct] email:nil licenseCode:nil activationStatusCompletion:^(PADActivationState activationState) {
            [self.view.window close];
        }];
    }
}


#pragma mark - Paddle helper methods

- (PADProduct *)mainProduct
{
    PADProduct *product = [[PADProduct alloc] initWithProductID:PAD_SDK_PRODUCT_ID
                                                    productType:PADProductTypeSDKProduct
                                                  configuration:[AppConfig configurationForMainProduct]];
    product.delegate = self;
    product.localizedTrialText = NSLocalizedStringFromTableInBundle(@"Thank you for trying out the Random Number Generator 500", nil, [NSBundle mainBundle], nil);
    return product;
}

- (PADProduct *)secondProduct
{
    PADProduct *product = [[PADProduct alloc] initWithProductID:PAD_SDK_SECOND_PRODUCT_ID
                                                    productType:PADProductTypeSDKProduct
                                                  configuration:[AppConfig configurationForSecondProduct]];
    product.delegate = self;

    return product;
}

@end

File: Swift/macOS SDK v4 Swift Sample/DeactivateViewController.swift
//
//  DeactivateViewController.swift
//  macOS SDK v4 Swift Sample
//
//  Created by Louis Harwood on 27/05/2021.
//  Copyright © 2021 Paddle. All rights reserved.
//

import Foundation
import Cocoa
import Paddle

class DeactivateViewController: NSViewController, PADProductDelegate {
    @IBOutlet var productSelection: NSPopUpButton!
    
    override func viewDidLoad() {
        if mainProduct().activated {
            productSelection.item(at: 0)?.isEnabled = false
        }
        
        if secondProduct().activated {
            productSelection.item(at: 1)?.isEnabled = false
        }
    }
    
    @IBAction func cancelButton(sender: NSButton) {
        view.window?.close()
    }
    
    @IBAction func nextButton(sender: NSButton) {
        if productSelection.indexOfSelectedItem == 0 {
            Paddle.sharedInstance()?.showLicenseActivationDialog(for: mainProduct(), email: nil, licenseCode: nil, activationStatusCompletion: { activationState in
                self.view.window?.close()
            })
        } else {
            Paddle.sharedInstance()?.showLicenseActivationDialog(for: secondProduct(), email: nil, licenseCode: nil, activationStatusCompletion: { activationState in
                self.view.window?.close()
            })
        }
    }
    
    
    func mainProduct() -> PADProduct {
        let product = PADProduct.init(productID: AppConfig.PAD_SDK_PRODUCT_ID, productType: PADProductType.sdkProduct, configuration: AppConfig.configurationForMainProduct())
        product?.delegate = self
        
        return product!
    }
    
    func secondProduct() -> PADProduct {
        let product = PADProduct.init(productID: AppConfig.PAD_SDK_SECOND_PRODUCT_ID, productType: PADProductType.sdkProduct, configuration: AppConfig.configurationForSecondProduct())
        product?.delegate = self
        
        return product!
    }
}
